extensions/GitExtension/FileExplorerGitIntegration/Models/GitCommandRunner.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using DevHome.Common.Helpers;
+using DevHome.Common.Services;
+using FileExplorerGitIntegration.Helpers;
+using Microsoft.Win32;
+using Microsoft.Windows.DevHome.SDK;
+using Serilog;
+using Windows.Storage;
+
+namespace FileExplorerGitIntegration.Models;
+
+public class GitCommandRunner : IDisposable
+{
+    public GitExecutableConfigOptions GitExecutableConfigOptions { get; set; }
+
+    private readonly FileService fileService;
+
+    private string GitExeInstallPath { get; set; } = string.Empty;
+
+    private readonly FileSystemWatcher fileWatcher;
+
+    private readonly ILogger log = Log.ForContext<GitCommandRunner>();
+
+    public GitCommandRunner(string? path)
+    {
+        if (RuntimeHelper.IsMSIX)
+        {
+            GitExecutableConfigOptions = new GitExecutableConfigOptions
+            {
+                GitExecutableConfigFolderPath = ApplicationData.Current.LocalFolder.Path,
+            };
+        }
+        else
+        {
+            GitExecutableConfigOptions = new GitExecutableConfigOptions
+            {
+                GitExecutableConfigFolderPath = path ?? string.Empty,
+            };
+        }
+
+        fileService = new FileService();
+        ReadInstallPath();
+
+        fileWatcher = new FileSystemWatcher(GitExecutableConfigOptions.GitExecutableConfigFolderPath, GitExecutableConfigOptions.GitExecutableConfigFileName);
+        fileWatcher.NotifyFilter = NotifyFilters.LastWrite;
+        fileWatcher.Changed += OnFileChanged;
+        fileWatcher.EnableRaisingEvents = true;
+        log.Debug("FileSystemWatcher initialized for configuration file");
+    }
+
+    private void OnFileChanged(object sender, FileSystemEventArgs args)
+    {
+        if (args.Name == GitExecutableConfigOptions.GitExecutableConfigFileName)
+        {
+            ReadInstallPath();
+        }
```

Comment:
outside of the extension process, what other processes are expected to modify this file/

Reply:
No other external processes are expected to modify this file. Thanks!

Reply:
Then we don't need a file watcher, when the process makes changes it could signal itself through an interface to reload.

Reply:
Goot point. If the Git extension's config UI is hosted in that process, then that's the only place we'd modify it. I hadn't made that first assumption, but that seems likely.

Reply:
Modified in latest commit. Thanks!